code/obj/item/device/pda2/smallprogs.dm

```diff
@@ -1108,7 +1108,9 @@ Using electronic "Detomatix" SELF-DESTRUCT program is perhaps less simple!<br>
 			playsound(src.master, 'sound/machines/printer_thermal.ogg', 50, 1)
 			SPAWN(3 SECONDS)
 				var/obj/item/paper/p = new /obj/item/paper
-				p.set_loc(get_turf(src.master))
+				if (ishuman(usr))
```

Comment:
This doesn't need an `ishuman` check, `put_in_hand_or_drop` is defined on `/mob` so `usr.put_in_hand_or_drop()` should be safe.